### PR TITLE
5 - Issue #96 Merged swagger generated classes

### PR DESCRIPTION
**Pull request by rg911**
- Auto generated api client typescript code by using Swagger CodeGen (typescript-node) template. 
- Updated http repositories to use new generated class.

** Dependant: 
- https://github.com/nemtech/nem2-sdk-typescript-javascript/pull/168

### CODE REVIEW

**Comment by rg911** [comment]
> you are missing the DTO models under infrastructure/model ?

Everything is in the ApiClient.ts class. Typescript Generator put everything into 1 file.

File: README.md [hunk 1]
```diff
@@ -78,6 +78,97 @@ Important versions listed below. Refer to the [Changelog](CHANGELOG.md) for a fu
...
+
+Following command can be used to generate DTOs and Api clients for the [nem2-sdk-typescript-javascript](https://github.com/nemtech/nem2-sdk-typescript-javascript) :
+
+1.
+    ```bash
+    $ git clone git@github.com:nemtech/nem2-docs
+    $ cd nem2-docs && mkdir sdks && cd sdks
+    $ cp ../source/resources/collections/swagger.yaml .
+    ```
+ 2. Copy the `templates` folder from {nem2-sdk-typescript-javascript}/infrustructure/ into `sdk` folder
```
Comment (evias) [inline README.md]: type `infrastructure`

File: README.md [hunk 2]
```diff
@@ -78,6 +78,97 @@ Important versions listed below. Refer to the [Changelog](CHANGELOG.md) for a fu
 - [0.10.1-beta](CHANGELOG.md#v0101-beta) - **Alpaca compatible** 07.2018
 - [0.9.5](CHANGELOG.md#095-27-Jun-2018) - **Alpaca compatible** 07.2018
 
+## Notes on generation of catapult-rest DTO and API client
+
+Following command can be used to generate DTOs and Api clients for the [nem2-sdk-typescript-javascript](https://github.com/nemtech/nem2-sdk-typescript-javascript) :
+
+1.
```
Comment (evias) [inline README.md]: add title "Download latest NEM2 swagger definition"

File: README.md [hunk 3]
```diff
@@ -78,6 +78,97 @@ Important versions listed below. Refer to the [Changelog](CHANGELOG.md) for a fu
...
+    ```bash
+    $ git clone git@github.com:nemtech/nem2-docs
+    $ cd nem2-docs && mkdir sdks && cd sdks
+    $ cp ../source/resources/collections/swagger.yaml .
+    ```
+ 2. Copy the `templates` folder from {nem2-sdk-typescript-javascript}/infrustructure/ into `sdk` folder
+
+3.
```
Comment (evias) [inline README.md]: Add title: "Download OpenAPI generator"

File: README.md [hunk 4]
```diff
@@ -78,6 +78,97 @@ Important versions listed below. Refer to the [Changelog](CHANGELOG.md) for a fu
...
+    ```bash
+    $ brew install openapi-generator
+    $ openapi-generator generate -i ./swagger2.yaml -g typescript-node -t templates/ -o ./nem2-ts-sdk/ && rm -R nem2-ts-sdk/test
+    ```
+    ** Note openapi-generator is also available on docker. (`https://hub.docker.com/r/openapitools/openapi-generator`)
+4. As the generator doesn't recognize `enum` type alias, we need to manually move enum classes in to the `enumsMap` list.
```
Comment (evias) [inline README.md]: Change title and move line in paragraph: "Fix enum type definitions"

**Pull request merged**